madmom/features/beats.py

```diff
@@ -390,6 +390,51 @@ def recursive(position):
     return np.array(positions)
 
 
+def detect_beats_with_ccf(activations, interval, look_aside=0.2):
```

Comment:
The problem you are facing here is the fact that many musical pieces simply don't have a constant tempo.

Although you're allowing some deviation from a constant tempo, this is only considered for each beat individually, i.e. the tempo remains stable and only some jitter is allowed. If one beat is a bit earlier (faster), the position of the next is not based on the current beat, but rather on the previously pre-computed positions.

`detect_beats` on the other hand allows also some deviation from the main tempo, but the position of the next beat is based on this position rather than the pre-computed ones. This makes this method more flexible, but also slower.

If `detect_beats_with_ccf` performs inferior to `detect_beats` on most datasets I don't see a point for including it, even if it is faster. I'd rather speed up `detect_beats` if speed is really an issue.